julia-package/brainflow/src/c_interfaces.jl

```diff
@@ -1,3 +1,39 @@
+using Pkg
+using Pkg.Artifacts
+using SHA
+using Tar
+
+function download_brainflow_artifact()
+
+    url = brainflow_url()
+
+    download_path = "$(tempname())-download.tar"
+    download(url, download_path)
+
+    brainflow_hash = create_artifact() do artifact_dir 
+        # Pkg.PlatformEngines.unpack() gives errors for some users on Windows
+        Tar.extract(download_path, artifact_dir)
+    end
+
+    rm(download_path)
+
+    return brainflow_hash
+end
+
+function get_brainflow_artifact_path()
+    # we have an issue with unpack() on Windows, so wrote a custom download
+    # https://discourse.julialang.org/t/unable-to-automatically-install-artifact/51984/2
+    artifacts_toml = find_artifacts_toml(@__DIR__) # is there a better way? this makes brainflow non-relocatable I believe.
```

Comment:
it seems like it always gets libs from the release page instead local folder. I think we can add a check like 

```
if (linux)
    file_name = './lib/libBoardController.so'
if (win)
    file_name = '.\\lib\\BoardController.dll'
if (mac)
    file_name = './lib/libBoardController.dylib'

if (!exists(file_name))
    # download artifacts
```

Will it work?

Reply:
No actually, if the artifact exists in the local folder it is not downloaded:

```
    if artifact_exists(brainflow_hash)
        return artifact_path(brainflow_hash)
```

Note however that we now move away from the `./lib/` folder into a Julia maintained artifact folder

Reply:
"Note however that we now move away from the ./lib/ folder into a Julia maintained artifact folder"
Yeah I realized it, can we mix these 2 options? I mean if there are local libs load them if there are not load artifacts

Reply:
or maybe there is a better way to achieve it

Reply:
hmm, yeah we could do this. But what to use in the CI then? Use the develop ./lib path, or the automatically installed libraries from the release url? I prefer to test what the user does (so use the released libraries), but then CI fails while developing until you release the libraries officially. Because if you use ./lib in CI, then after releasing you'll have to manually update CI to check if the automatic download process works on all platforms.

This is why I still prefer a more decoupled deployment process. First work on the c++, when that is fine and tested you (internally) deploy the libraries, then use the new url in the Julia/MATLAB/etc bindings, then we update those bindings, then release those bindings. 

I guess you do internal development only on Python right now anyway? Where you quickly compile c++ libraries, then check how they feel in Python, then update c++ again to improve, repeat.

Reply:
"But what to use in the CI then? Use the develop ./lib path, or the automatically installed libraries from the release url?"
It should be two different cases and two different CI runs, in existing workflows we need to test local libs(and there we already precompile them), because existing workflow mostly for developers. I have a task to test released packages(https://github.com/brainflow-dev/brainflow/issues/167) and for julia it will test package from Artifacts

Reply:
"I guess you do internal development only on Python right now anyway? Where you quickly compile c++ libraries, then check how they feel in Python, then update c++ again to improve, repeat."

No, its mostly in Cpp but for new functions need to update all bindings at once, so it includes Julia too. Its way more better to do it in single git commit\PR

Reply:
Also, another reason to add the ability to use local libs - performace. Precompiled libraries dont have openmp(because users will need to install system packages and other dependencies and some of them dont need it). So, existing policy is that precompiled libs have no OpenMP parallelization and if you want to increase perf - compile from the source with openmp enabled.

In a long term this policy more likely will be changed and I will split it by two packages - with and without openmp

Reply:
very well, if we get 2 CIs we can split.
I'll add the ./lib back in then.